cluster/addons/metrics-server/metrics-server-deployment.yaml

```diff
@@ -54,17 +54,34 @@ spec:
         command:
         - /metrics-server
         - --metric-resolution=30s
-        # These are needed for GKE, which doesn't support secure communication yet.
-        # Remove these lines for non-GKE clusters, and when GKE supports token-based auth.
-        - --kubelet-port=10255
-        - --deprecated-kubelet-completely-insecure=true
+        - --kubelet-use-node-status-port
         - --kubelet-preferred-address-types=InternalIP,Hostname,InternalDNS,ExternalDNS,ExternalIP
```

Comment:
I am not too familiar with the test environment, but maybe we couldn't access the kubelet metrics via Hostname? If that's the case we might want to update the preference list. We have `InternalIP,ExternalIP,Hostname` upstream to align with kubeadm.

Reply:
it seems this change give us more light
> I0719 14:08:53.355] E0719 14:05:36.133995       1 server.go:132] unable to fully scrape metrics: [unable to fully scrape metrics from node e2e-27cbc2e916-a7d53-master: unable to fetch metrics from node e2e-27cbc2e916-a7d53-master: Get "https://10.40.0.2:10250/stats/summary?only_cpu_and_memory=true": x509: cannot validate certificate for 10.40.0.2 because it doesn't contain any IP SANs, unable to fully scrape metrics from node e2e-27cbc2e916-a7d53-minion-group-mg0c: unable to fetch metrics from node e2e-27cbc2e916-a7d53-minion-group-mg0c: Get "https://10.40.0.4:10250/stats/summary?only_cpu_and_memory=true": x509: cannot validate certificate for 10.40.0.4 because it doesn't contain any IP SANs, unable to fully scrape metrics from node e2e-27cbc2e916-a7d53-minion-group-z50r: unable to fetch metrics from node e2e-27cbc2e916-a7d53-minion-group-z50r: Get "https://10.40.0.5:10250/stats/summary?only_cpu_and_memory=true": x509: cannot validate certificate for 10.40.0.5 because it doesn't contain any IP SANs, unable to fully scrape metrics from node e2e-27cbc2e916-a7d53-minion-group-qkbw: unable to fetch metrics from node e2e-27cbc2e916-a7d53-minion-group-qkbw: Get "https://10.40.0.3:10250/stats/summary?only_cpu_and_memory=true": x509: cannot validate certificate for 10.40.0.3 because it doesn't contain any IP SANs]

Reply:
I think we should be able to skip the certificates by passing `--kubelet-insecure-tls` to metrics-server.

Reply:
let me try that

Reply:
it works, thanks

Reply:
it worked